### PR TITLE
images

### PR DESCRIPTION
**Pull request by kirillova-ak5**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,59 @@
...
+
+### Photometric
+- photo must be in *\*.jpg* format
+- camera direction can be rotated no more then *15&deg;* from surface macronormal (vertical in common case)
+- illumination and color of objects should be such that they contrast with surface
+    - object border at least 10 units darker than surface (if lightness is sum of components in R8G8B8 colorspace)
+- other photometric parameters are not specified, but photo still should meet other types of requirements
+
+
+## Chosen set of items
```
Comment (Tatiana655) [inline README.md:54]: Dataset is good. I have a question about a plush toy. Are you going to position as in the photos or permanent position? I ask because in some of the photos the positions of some parts of the toy (tail) are different.

File: test/test_document.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+Checks that program can work with one object and such polygon, where its clearly seen, can we plase, or not. 
+![test 0](/test/img/0.jpg "test 0")
+Answer: true
+
+![test 1](/test/img/1.jpg "test 1")
+Answer: false
+
+
+### Precision test
+There are two two similar polygons(triangles) and their size is is close to the corner case of bounding triangle of object.
```
Comment (Tatiana655) [inline test/test_document.md]: two :)

File: README.md [hunk 3]
```diff
@@ -1 +1,59 @@
...
+- objects and polygon must be placed on white (or light) surface
+- objects are from training dataset
+- objects must not collide with each other
+    - at least 4 pixels between objects must contain only pixels of surface, no objects pixels
+- objects must not collide with polygon
+    - at least 4 pixels between objects and polygon must contain only pixels of surface, no objects or polygon pixels
+- objects and polygon must be entirely contained in a photo
+    - at least 4 pixels on boards of photo must contain only pixels of surface, no objects or polygon pixels
+- objects must be placed outside of the polygon
+- polygon must be black (or dark)
```
Comment (Tatiana655) [inline README.md]: mb drawn using dark marker, because only its borders are dark

File: README.md [hunk 4]
```diff
@@ -1 +1,59 @@
...
+    - object border at least 10 units darker than surface (if lightness is sum of components in R8G8B8 colorspace)
+- other photometric parameters are not specified, but photo still should meet other types of requirements
+
+
+## Chosen set of items
+Chosen set of items and white surface can be viewed [here](https://github.com/kirillova-ak5/intelligent_placer/blob/develop/item_set/item_set_document.md)
+
+
+## Training dataset
+Training dataset with correct answers, comments and corner cases can be viewed [here](https://github.com/kirillova-ak5/intelligent_placer/blob/develop/test/test_document.md)
```
Comment (Oleg595) [inline README.md:59]: There are shadows in many photos. Can your algorithm recognise this as object?

File: README.md (hunk 1 above)
Comment (kirillova-ak5) [reply README.md:54]: position does not have to be permanent from one test to another, but while place program threat object as solid, so position in of tail relative to the body in hypothetical placements is same as input photo

File: test/test_document.md (hunk 2 above)
Comment (kirillova-ak5) [reply test/test_document.md]: fixed

File: README.md (hunk 3 above)
Comment (kirillova-ak5) [reply README.md]: fixed

File: README.md (hunk 4 above)
Comment (kirillova-ak5) [reply README.md:59]: i want to believe that it can

File: README.md [hunk 5]
```diff
@@ -1 +1,59 @@
...
+And if exist disposition like this:
+![test 10 proof](/item_set/a.jpg "test 10 proof")
+where all objects placed in polygon, programm answer true, otherwise - false.
+
+Now this is just educational task, so programm can receive only photos of objects, which were in training dataset.
+
+## Formal statement of the problem
+### General
+- programm gets input with path to photo of objects and polygon(all objects and polygon on one photo)
+- programm must answer and stop working in no more than *5 minutes*
```
Comment (Shaots) [inline README.md:15]: очень круто, что ты прибавила ограничение на время. это увеличивало сложность выполнения алгоритма.

File: test/map.csv [hunk 6]
```diff
@@ -0,0 +1,31 @@
+image path,rezult,comment
```
Comment (Shaots) [inline test/map.csv:1]: result

**Comment by Shaots** [comment]
Постановка задачи хорошая, с учетом многообразных деталей как требование к данным: неодносвязная область, так и к программе: время выполнения, warning. Подробно рассмотрены всякие случаи в сборе данных.

File: README.md [hunk 7]
```diff
@@ -1 +1,59 @@
...
+- objects and polygon must be placed on white (or light) surface
+- objects are from training dataset
+- objects must not collide with each other
+    - at least 4 pixels between objects must contain only pixels of surface, no objects pixels
+- objects must not collide with polygon
+    - at least 4 pixels between objects and polygon must contain only pixels of surface, no objects or polygon pixels
+- objects and polygon must be entirely contained in a photo
+    - at least 4 pixels on boards of photo must contain only pixels of surface, no objects or polygon pixels
+- objects must be placed outside of the polygon
+- polygon must have black (or dark) borders and transparent interior (common case is drowen on paper with marker)
```
Comment (MihailTchekmarev) [inline README.md:44]: The width of the polygon segments is not specified. Since it's can be a drawn figure, I believe it will be a good thing to add a minimal requiered width.

File: test/test_document.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+Stress test with a lot of objects(all) and wery carefully placement. For true answer i attach picture-proof, because its hard to believe that all objeckts really can be placed on A4.
+![test 10](/test/img/10.jpg "test 10")
+![test 10 proof](/item_set/a.jpg "test 10 proof")
+Answer: true
+
+![test 11](/test/img/11.jpg "test 11")
+Answer: false
+
+
+### Corner cases
```
Comment (MihailTchekmarev) [inline test/test_document.md:65]: All test cases seem to be representative. I think it will also be nice to see how the program can deal with low resolution or blurred pictures,

File: README.md (hunk 7 above)
Comment (MihailTchekmarev) [reply README.md:44]: Also a maximum number of polygon corners can be specified, since a curved line can be mistakingly recognised as a polygon with large amount of links. (Unless such behaviour is acceptable)
